storage/store.go

```diff
@@ -0,0 +1,37 @@
+package storage
+
+import (
+	"context"
+
+	"github.com/odpf/stencil/models"
+)
+
+// Store is the interface that all database objects must implement.
+type Store interface {
+	// ListSnapshots returns a list of snapshots that match the query fields.
+	ListSnapshots(ctx context.Context, query *models.Snapshot) ([]*models.Snapshot, error)
+
+	// UpdateSnapshotLatestVersion updates the latest version of the snapshot.
+	UpdateSnapshotLatestVersion(ctx context.Context, snapshot *models.Snapshot) error
+
+	// GetSnapshotByFields returns the snapshot with the given fields.
+	GetSnapshotByFields(ctx context.Context, namespace, name, version string, latest bool) (*models.Snapshot, error)
+
+	// GetSnapshotByID returns the snapshot with the given id.
+	GetSnapshotByID(ctx context.Context, id int64) (*models.Snapshot, error)
+
+	// ExistsSnapshot returns true if the snapshot exists.
+	ExistsSnapshot(ctx context.Context, snapshot *models.Snapshot) bool
+
+	// CreateSnapshot creates a new snapshot.
+	CreateSnapshot(ctx context.Context, snapshot *models.Snapshot) error
+
+	// DeleteSnapshot deletes the snapshot.
+	DeleteSnapshot(ctx context.Context, snapshot *models.Snapshot) error
+
+	// GetPBFiles returns the protobuf file with the given name.
+	GetPBFiles(ctx context.Context, snapshot *models.Snapshot, names []string) ([][]byte, error)
```

Comment:
Let's call this some thing else instead of `GetPBFiles`? Since this is actually getting a `FileDescriptorSet` can we call it some thing of that type?

Reply:
Yes, plan is to call this `GetSchema`.